roseau/load_flow/models/core.py

```diff
@@ -107,22 +133,61 @@ def current(self) -> complex:
         """
         raise NotImplementedError
 
+    @classmethod
+    def from_dict(cls, data: JsonDict) -> "PotentialRef":
+        return cls(data["id"], data["element"], phase=data.get("phases"))
+
+    def to_dict(self) -> JsonDict:
+        from roseau.load_flow.models.buses import Bus  # TODO refactor potential ref and ground
+
+        res = {"id": self.id}
+        e = self.connected_elements[0]
+        if isinstance(e, Bus):
+            res["bus"] = e.id
+            res["phases"] = self.phase
+        elif isinstance(e, Ground):
+            res["ground"] = e.id
+        else:
+            assert False, f"Unexpected element type {type(e).__name__}"
+        return res
+
 
 class Ground(Element):
-    """This element defines the ground."""
+    """This element defines the ground.
+
+    Only buses and lines that have shunt components can be connected to a ground.
+
+    1. Connecting to a bus:
+
+       To connect a ground to a bus on a given phase, use the :meth:`Ground.connect_to_bus` method.
+       This method lets you specify the bus to connect to as well as the phase of the connection.
+       If the bus has a neutral and the phase is not specified, the ground will be connected to the
+       neutral, otherwise, an error will be raised because the phase is needed.
+
+    2. Connecting to a line with shunt components:
+
+       To connect a ground to a line with shunt components, pass the ground object to the
+       :class:`Line` constructor. Note that the ground connection is mandatory for shunt lines.
+    """
 
     allowed_phases = frozenset({"a", "b", "c", "n"})
 
-    def __init__(self, **kwargs):
-        """Ground constructor."""
-        super().__init__(**kwargs)
-        self.phase: Optional[str] = None
+    def __init__(self, id: Id, **kwargs: Any) -> None:
+        """Ground constructor.
+
+        Args:
+            id:
+                A unique ID of the ground in the network grounds.
+        """
+        super().__init__(id, **kwargs)
+        self.phases: dict[Id, str] = {}
+        """A map of bus id to phase connected to this ground."""
 
     def __repr__(self) -> str:
-        return f"{type(self).__name__}()"
+        return f"{type(self).__name__}(id={self.id!r})"
 
-    def connect(self, bus: "Bus", phase: str = "n"):
-        """Connect the ground to the bus neutral.
+    def connect_to_bus(self, bus: "Bus", phase: str = "n"):
+        """Connect the ground to a bus on the given phase.
```

Comment:
I realize that there are several problems here that I did not expect:

1. A `Ground` has a `connect` method and a `connect_to_bus` method. That is confusing…

    *First proposal*: I would suggest you to add a `**kwargs` in the signature of `Element.connect` and then to overload the `connect` method here by declaring a `phase` argument (default to `None`). 

    The element can be 
      * a Bus and phase is required (if phase is None, take "n")
      * a Ground and phase must not be precised.
      * a Line (for the shunt) and phase is not necessary
    You can call the super method inside the implementation of this method.

    ⇾ First proposal seems uselessly complicated...

2. There is no check in the `connect` method. I can do `bus1.connect(bus2)`. It does not really have a meaning. 
    Moreover, in the actual code, you use `self.connect(ground)` in the `Line` constructor while the `ground.connect(self)` would be forbidden without any difference between `connect` and `connect_to_bus`…

Second proposal: The method `connect` of `Element` is renamed into `_connect` to avoid user to use it directly. In the `Ground` class, you rename the `connect_to_bus` into `connect` (public method. Keep the signature of the `connect_to_bus` method ⇾ `(bus, phase="n")`). In this method, use the `_connect` as you have done.

The user will not be able to `connect` elements except a ground to a bus' phase. The `disconnect` method remains public for the moment.


*NB for later:*
Later, the disconnect method may also become protected. We need to think at a clear interface, as there are methods `add_element` and `remove_element` in the `ElectricalNetwork` too... At some point, any element may be aware of the electrical network it belongs to (or if it does not belong to a network). Calling `bus.disconnect(load)` will also trigger something in the electrical network they belong too to ensure that the underlying C++ network model is declared as invalid and the results of all elements of this network are outdated (by logging a warning for instance). It may also ensure that after the call to any `update_*` method, the results are also declared as invalid.